### PR TITLE
fix: bundle generation

### PR DESCRIPTION
**Pull request by hdamarcelo**
- Add skipRange
- Use correct camel-k and strimzi versions
- Use sha256 for camel an strimzi

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -64,7 +59,7 @@ bundle/cos-fleetshard-operator-camel: operator-sdk
 	# yq -i 'del(.spec.install.spec.deployments[].label)' \
 	# 	$(ADDON_PATH)/cos-fleetshard-operator-camel/$(ADDON_VERSION)/manifests/cos-fleetshard-operator-camel.clusterserviceversion.yaml
 
-	yq -i '.spec.replaces="$(CAMEL_BUNDLE_NAME).$(PREVIOUS_TAG)"' \
+	yq -i '.metadata.annotations."olm.skipRange"=">=$(SKIP_RANGE_START) <$(ADDON_VERSION)"' \
```
Comment (lburgazzoli) [inline Makefile:62]: @valdar is this what we should do by default ? I thought this were a workaround

File: hack/bundle.sh [hunk 2]
```diff
@@ -25,39 +25,81 @@ if [ ! -d "${DIR_OVERLAY}" ]; then
...
+function bundle_external_operators() {
+  local kustomization=$1
+  local container_image=''
+  local version=''
+
+  container_image=$(printf '%s\n' "$kustomization" | yq 'select(.kind=="Deployment").spec.template.spec.containers[0].image')
+  version=$(printf '%s' "$container_image" | cut -f 2 -d ':' | tr '[:upper:]' '[:lower:]')
+  version=${version:0:25}
+  container_image_name=$(printf '%s' "$container_image" | cut -f 1 -d ':')
+  container_image_sha256=$(skopeo inspect "docker://$container_image" --format "{{.Digest}}")
```
Comment (lburgazzoli) [inline hack/bundle.sh:37]: Ideally this should happen in the cos-fleetshard pipeline [1][2] so installing the service with kustomize would produce (almost) the same result 

[1] https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/blob/main/.github/workflows/build-tag.yaml#L75-L89
[2] https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/blob/main/.github/workflows/build-main.yaml#L91-L102

Comment (rinaldodev) [reply hack/bundle.sh:37]: Ideally, yes. But we need the tag in this step to be able to extract the version name to OLM, which requires semantic versioning and is extracted from the tag name. Any ideas on how to overcome this for now?

Comment (lburgazzoli) [reply hack/bundle.sh:37]: I think can grab such info from from an annotation/label. wdyt ?

**Review by lburgazzoli — APPROVED** [review]

**Pull request merged**